.github/workflows/test.yaml

```diff
@@ -3,6 +3,51 @@ on:
   pull_request:
     branches: [ "master", "release/**" ]
 jobs:
+  snap-builds:
+    runs-on: ubuntu-16.04
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Cache snapd snap build status
+      id: cache-snapd-build-status
+      uses: actions/cache@v1
+      with:
+        path: "${{ github.workspace }}/.test-results"
+        key: "${{ github.run_id }}-${{ github.job }}-results"
+    - name: Check cached snap build
+      id: cached-results
+      run: |
+          CACHE_RESULT_STAMP="${{ github.workspace }}/.test-results/snap-build-success"
+          echo "::set-env name=CACHE_RESULT_STAMP::$CACHE_RESULT_STAMP"
+          if [ -e "$CACHE_RESULT_STAMP" ]; then
+              has_cached_snap=0
+              while read name; do
+                  has_cached_snap=1
+                  # bring back artifacts from the cache
+                  cp -v "$name" "${{ github.workspace }}"
+              done < <(find "$(dirname $CACHE_RESULT_STAMP)" -name "*.snap")
+              if [ "$has_cached_snap" = "1" ]; then
+                  # we have restored an artifact from the cache
+                  echo "::set-output name=already-ran::true"
+              fi
```

Comment:
heh, this ended up being quite a bit more complicated than I thought, glad you figured it out though